### ISSUE TITLE
[Feature request] RSS-Only Mode

### ISSUE THREAD

**Issue by RandomErrorMessage** [iv-org/invidious#2819]
I use Invidious almost exclusively as an RSS proxy for YouTube. I think Invidious does a really good job generating RSS feeds compared to the official feeds. I would like to be able to run my Invidious instance in a mode where the front-end is disabled, and the video links in the RSS feed point directly at YouTube. I personally use mpv+yt-dlp as my viewer, and having links point to the invidious front-end is just extra clicking for me.

This would also allow people to run light-weight public instances with less concern about being throttled. A half-way step for people who want to maintain channel subscriptions in a decentralized manner, but want to use a non-Invidious front-end.

Thanks for considering my feature request.

**Comment by unixfox** [comment]
You only want to have the backend with the RSS feeds endpoint then?

**Comment by RandomErrorMessage** [comment]
Yes, I only want the GET /feed/channel/[CHANNEL_ID] RSS/Atom feature. I'd like the feed to return links directly to the YouTube video page.

**Comment by unixfox** [comment]
You can build a version of Invidious without a frontend if that's what you are asking:
```
crystal build -Dapi_only src/invidious.cr
```

You will still need a database because the RSS feed is built using some data from the database though.

Is this what you want?

**Comment by RandomErrorMessage** [comment]
I tried with -Dapi_only and it resulted in the RSS feed path going into an endless redirect. I'd like `HTTP GET /feed/channel/[CHANNEL_ID]` to function, and I'd like the videos in the RSS feed link to youtube instead of the invidious instance itself. To be able to disable other unnecessary features would be nice in this use-case.

**Comment by unixfox** [comment]
Mmh that's correct, sorry I thought this would leave the feed endpoint at least.

**Comment by Patraskon** [comment]
Is it possible to make rss from a search query?

**Comment by unixfox** [comment]
> Is it possible to make rss from a search query?

Can you open another github issue? Your request is different from this current github issue.
